packages/core/src/internal/reconciliation/helpers/reconcile-strategy.ts

```diff
@@ -0,0 +1,70 @@
+import {
+  ContractAtFuture,
+  ContractCallFuture,
+  ContractDeploymentFuture,
+  LibraryDeploymentFuture,
+  NamedArtifactContractAtFuture,
+  NamedArtifactContractDeploymentFuture,
+  NamedArtifactLibraryDeploymentFuture,
+  ReadEventArgumentFuture,
+  SendDataFuture,
+  StaticCallFuture,
+} from "../../../types/module";
+import {
+  CallExecutionState,
+  ContractAtExecutionState,
+  DeploymentExecutionState,
+  ReadEventArgumentExecutionState,
+  SendDataExecutionState,
+  StaticCallExecutionState,
+} from "../../execution/types/execution-state";
+import {
+  ReconciliationContext,
+  ReconciliationFutureResultFailure,
+} from "../types";
+import { fail } from "../utils";
+
+export function reconcileStrategy(
+  future:
+    | NamedArtifactContractDeploymentFuture<string>
+    | ContractDeploymentFuture
+    | NamedArtifactLibraryDeploymentFuture<string>
+    | LibraryDeploymentFuture
+    | NamedArtifactContractAtFuture<string>
+    | ContractAtFuture
+    | ContractCallFuture<string, string>
+    | StaticCallFuture<string, string>
+    | SendDataFuture
+    | ReadEventArgumentFuture,
+  exState:
+    | DeploymentExecutionState
+    | CallExecutionState
+    | StaticCallExecutionState
+    | ContractAtExecutionState
+    | SendDataExecutionState
+    | ReadEventArgumentExecutionState,
+  context: ReconciliationContext
+): ReconciliationFutureResultFailure | undefined {
+  const storedStrategyName = exState.strategy;
+  const newStrategyName = context.strategy;
+
+  if (storedStrategyName !== newStrategyName) {
+    return fail(
+      future,
+      `Strategy changed from "${storedStrategyName}" to "${newStrategyName}"`
+    );
+  }
+
+  const storedStrategyConfig = exState.strategyConfig;
+  const newStrategyConfig = context.strategyConfig;
+
+  const isEqual = require("lodash/isEqual") as typeof import("lodash/isEqual");
```

Comment:
@zoeyTM what is the thinking behind the dynamic load here? Is it the weight of loading lodash?

Reply:
The weight of loading lodash, yes, but mostly just mimicking the way it was imported in the `reconcile-arguments` function.

Reply:
I think this makes sense. If we had the slow imports plugins that luist built, it would have forced you to do this.